pkg/build/gobuild_test.go

```diff
@@ -437,6 +437,7 @@ func TestGoBuildNoKoData(t *testing.T) {
 		WithBaseImages(func(context.Context, string) (name.Reference, Result, error) { return baseRef, base, nil }),
 		withBuilder(writeTempFile),
 		withSBOMber(fauxSBOM),
+		WithPlatforms("all"),
```

Comment:
I'm confused a bit why this addition is needed for tests. I would have expected `random.Image` to produce a `linux/amd64` image by default, and for `NewGo` to assume that platform if none is provided, so that platform matcher check you added wouldn't be necessary.

Does this change require callers of the Go API to explicitly pass `WithPlatforms`?

Reply:
`random.Image` has no platform (or any config at all), so you get this somewhat odd error

```
gobuild_test.go:448: Build() = base image platform "" does not match desired platforms []
```

It looks like from `empty.Image` it just gets an empty Platform struct:
```go
*v1.Platform{Architecture: "", OS: "", OSVersion: "", OSFeatures: []string len: 0, cap: 0, nil, Variant: "", Features: []string len: 0, cap: 0, nil}
```

The CLI calls [gobuildOptions](https://github.com/benmoss/ko/blob/8211102f214d6f024797db637bdeead2e871aa28/pkg/commands/resolver.go#L50) which handles the defaulting of desired platforms.

My understanding from reading over the code is that the Go API (`NewGo`) doesn't do any defaulting, so if the user didn't set any desired platforms the current behavior would be:
- if given an image index/manifest list, [no platforms would match](https://github.com/benmoss/ko/blob/8211102f214d6f024797db637bdeead2e871aa28/pkg/build/gobuild.go#L985)
- if given an image manifest, it would [copy the platform](https://github.com/benmoss/ko/blob/8211102f214d6f024797db637bdeead2e871aa28/pkg/build/gobuild.go#L752-L758) from the base image

The tests were relying on that latter behavior, so the resulting image would also have a blank platform.